lib/src/js/value/color.dart

```diff
@@ -2,66 +2,366 @@
+            changedValue('chroma'),
+            changedValue('hue'),
+            changedValue('alpha'));
+        break;
+
+      case ColorSpace.a98Rgb:
+      case ColorSpace.displayP3:
+      case ColorSpace.prophotoRgb:
+      case ColorSpace.srgb:
+      case ColorSpace.srgbLinear:
+        changedColor = SassColor.forSpaceInternal(
+            spaceClass,
+            changedValue('red'),
+            changedValue('green'),
+            changedValue('blue'),
+            changedValue('alpha'));
+        break;
+
+      case ColorSpace.xyzD50:
+      case ColorSpace.xyzD65:
+        changedColor = SassColor.forSpaceInternal(spaceClass, changedValue('x'),
+            changedValue('y'), changedValue('z'), changedValue('alpha'));
+        break;
+
+      default:
+        throw "No space set";
+    }
+
+    return changedColor.toSpace(initialSpaceClass);
   });
 
   jsClass.defineGetters({
-    'red': (SassColor self) => self.red,
-    'green': (SassColor self) => self.green,
-    'blue': (SassColor self) => self.blue,
-    'hue': (SassColor self) => self.hue,
-    'saturation': (SassColor self) => self.saturation,
-    'lightness': (SassColor self) => self.lightness,
-    'whiteness': (SassColor self) => self.whiteness,
-    'blackness': (SassColor self) => self.blackness,
-    'alpha': (SassColor self) => self.alpha,
+    'red': (SassColor self) {
+      _emitColor4ApiDeprecation('red');
+      return self.red;
+    },
+    'green': (SassColor self) {
+      _emitColor4ApiDeprecation('green');
+      return self.green;
+    },
+    'blue': (SassColor self) {
+      _emitColor4ApiDeprecation('blue');
+      return self.blue;
+    },
+    'hue': (SassColor self) {
+      _emitColor4ApiDeprecation('hue');
+      return self.hue;
+    },
+    'saturation': (SassColor self) {
+      _emitColor4ApiDeprecation('saturation');
+      return self.saturation;
+    },
+    'lightness': (SassColor self) {
+      _emitColor4ApiDeprecation('lightness');
+      return self.lightness;
+    },
+    'whiteness': (SassColor self) {
+      _emitColor4ApiDeprecation('whiteness');
+      return self.whiteness;
+    },
+    'blackness': (SassColor self) {
+      _emitColor4ApiDeprecation('blackness');
+      return self.blackness;
+    },
+  });
+
+  jsClass.defineGetter('alpha', (SassColor self) => self.alpha);
+
+  jsClass.defineGetters({
+    'space': (SassColor self) => self.space.name,
+    'isLegacy': (SassColor self) => self.isLegacy,
+  });
+
+  jsClass.defineMethod('toSpace', (SassColor self, String space) {
+    if (self.space.name == space) {
+      return self;
+    }
+    ColorSpace spaceClass = ColorSpace.fromName(space);
+    return self.toSpace(spaceClass);
+  });
+
+  jsClass.defineMethod('isInGamut', (SassColor self, [String? space]) {
+    String spaceName = space ?? self.space.name;
+    ColorSpace spaceClass = ColorSpace.fromName(spaceName);
+    SassColor color = self.toSpace(spaceClass);
+    return color.isInGamut;
+  });
```

Comment:
Style nit: this uses a lot of intermediate variables for very small expressions. It would probably be cleaner as just

```suggestion
  jsClass.defineMethod('isInGamut', (SassColor self, [String? space]) =>
      self.toSpace(ColorSpace.fromName(space ?? self.space.name)).isInGamut);
```

You could also potentially factor out some of this into a `SassColor _toSpace(SassColor self, String? space)` function, since the logic gets re-used a lot.